### PR TITLE
SAN-24-Fixing-Date-and-Amount updating ConfirmationController and ConfirmationControllerTest

### PR DESCRIPTION
**Pull request by klyttle-ch**
updating ConfirmationController and ConfirmationControllerTest

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java [hunk 1]
```diff
@@ -49,13 +53,20 @@ public class ConfirmationController extends BaseController {
...
+
+    @Autowired
+    private PenaltyPaymentService penaltyPaymentService;
+
     @Autowired
     public ConfirmationController(CompanyService companyService,
-            PayablePenaltyService payablePenaltyService,
-            SessionService sessionService) {
+                                  PayablePenaltyService payablePenaltyService,
+                                  SessionService sessionService, PenaltyPaymentService penaltyPaymentSerivce) {
```
Comment (dgordon-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:65]: penaltyPaymentSerivce -- Service

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java [hunk 2]
```diff
@@ -113,17 +128,15 @@ public String getConfirmation(@PathVariable String companyNumber,
 
     private String setUpPaymentDateDisplay(PayableLateFilingPenalty payableLateFilingPenalty) {
         if (payableLateFilingPenalty.getPayment() != null) {
-            return StringUtils.isEmpty(payableLateFilingPenalty.getPayment().getPaidAt()) ? "" :
-                    OffsetDateTime.parse(payableLateFilingPenalty.getPayment().getPaidAt(), DateTimeFormatter.ISO_OFFSET_DATE_TIME)
-                            .format(DateTimeFormatter.ofPattern("d MMMM uuuu", Locale.UK));
+            return LocalDate.now()
+                    .format(DateTimeFormatter.ofPattern("d MMMM uuuu", Locale.UK));
```
Comment (dgordon-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:132]: @klai-ch -- is this correct? I know you added in the additional parsing for the date, and is the requirement just to pass back today's date (pretty sure it is).

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java [hunk 3]
```diff
@@ -88,6 +101,8 @@ public String getConfirmation(@PathVariable String companyNumber,
         try {
             PayableLateFilingPenalty payablePenalty = payablePenaltyService
                     .getPayableLateFilingPenalty(companyNumber, payableRef);
+            lateFilingPenalties = penaltyPaymentService.getLateFilingPenalties(companyNumber, penaltyRef);
```
Comment (dgordon-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:104]: This opens up question on whether we need to make this generic / not lfp specific or what we'd do in that instance. Need to have a think for future task around changing / refactoring this, but tbf that can come later.

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java [hunk 4]
```diff
@@ -113,17 +128,15 @@ public String getConfirmation(@PathVariable String companyNumber,
...
         return "";
     }
 
-    private String setUpPaymentAmountDisplay(PayableLateFilingPenalty payableLateFilingPenalty) {
-        if (payableLateFilingPenalty.getPayment() != null) {
-            return StringUtils.isEmpty(payableLateFilingPenalty.getPayment().getAmount()) ? "" :
-                    "£" + payableLateFilingPenalty.getPayment().getAmount() + " (no VAT is charged)";
+    private String setUpPaymentAmountDisplay(LateFilingPenalty payableLateFilingPenalty) {
+        if (payableLateFilingPenalty.getOriginalAmount() != null) {
+            return "£" + penaltyUtils.getFormattedOutstanding(payableLateFilingPenalty.getOriginalAmount()) + " (no VAT is charged)";
```
Comment (dgordon-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:139]: Would we add in the "£" to the html template instead? See viewPenalties.html for example. To keep it consistent, something like;

                    <tr class="govuk-table__row">
                        <td class="govuk-table__header">Amount</td>
                        <td class="govuk-table__cell" th:text="'£' + ${outstanding} + ' (no VAT is charged)'"></td>
                    </tr>

**Review by dgordon-ch — CHANGES_REQUESTED** [review]
Added some comments.

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java (hunk 4 above)
Comment (klyttle-ch) [reply src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:139]: I'll get this into the html instead

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java (hunk 2 above)
Comment (klai-ch) [reply src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:132]: Yes, the requirement is passing back today's date. I use the payableLateFilingPenalty.payment.paidAt (which is also today's date) in order to make good use the module payableLateFilingPenalty.payment and I don't know if there is any chances that it is not pass today's date in the future, so I just think it is more accurate.

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java [hunk 5]
```diff
@@ -88,6 +101,8 @@ public String getConfirmation(@PathVariable String companyNumber,
         try {
             PayableLateFilingPenalty payablePenalty = payablePenaltyService
```
Comment (klai-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:102]: I am thinking do we still need this payablePenalty or not, because we will be using lateFilingPenalty in stead.

Comment (klyttle-ch) [reply src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:102]: Still used in setUpPaymentDisplay to verify if .getPayment() is not null

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/test/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationControllerTest.java [hunk 6]
```diff
@@ -181,6 +191,8 @@ void getRequestStatusIsCancelled() throws Exception {
 
         when(mockPayablePenaltyService.getPayableLateFilingPenalty(COMPANY_NUMBER, PAYABLE_REF))
                 .thenReturn(PPSTestUtility.validPayableLateFilingPenalty(COMPANY_NUMBER, PENALTY_REF));
+        when(mockPenaltyPaymentService.getLateFilingPenalties(COMPANY_NUMBER, PENALTY_REF))
+                .thenReturn(List.of(new LateFilingPenalty()));
```
Comment (acallaghan-ch) [inline src/test/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationControllerTest.java:195]: Could use this instead:
`Collections.singletonList(new LateFilingPenalty())`

File: src/test/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationControllerTest.java [hunk 7]
```diff
@@ -113,13 +121,15 @@ void getRequestSuccessNullPenalty() throws Exception {
 
         PayableLateFilingPenalty mockPenalty = PPSTestUtility.validPayableLateFilingPenalty(COMPANY_NUMBER, PENALTY_REF);
         mockPenalty.setPayment(null);
+
         when(mockCompanyService.getCompanyProfile(COMPANY_NUMBER))
                 .thenReturn(PPSTestUtility.validCompanyProfile(COMPANY_NUMBER));
         when(mockPayablePenaltyService.getPayableLateFilingPenalty(COMPANY_NUMBER, PAYABLE_REF))
                 .thenReturn(mockPenalty);
+        when(mockPenaltyPaymentService.getLateFilingPenalties(COMPANY_NUMBER, PENALTY_REF))
+                .thenReturn(List.of(new LateFilingPenalty()));
```
Comment (acallaghan-ch) [inline src/test/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationControllerTest.java:130]: Could use this instead:
`Collections.singletonList(new LateFilingPenalty())`

File: src/test/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationControllerTest.java [hunk 8]
```diff
@@ -86,6 +92,8 @@ void getRequestSuccess() throws Exception {
                 .thenReturn(PPSTestUtility.validCompanyProfile(COMPANY_NUMBER));
         when(mockPayablePenaltyService.getPayableLateFilingPenalty(COMPANY_NUMBER, PAYABLE_REF))
                 .thenReturn(PPSTestUtility.validPayableLateFilingPenalty(COMPANY_NUMBER, PENALTY_REF));
+        when(mockPenaltyPaymentService.getLateFilingPenalties(COMPANY_NUMBER, PENALTY_REF))
+                .thenReturn(Collections.singletonList(new LateFilingPenalty()));
```
Comment (acallaghan-ch) [inline src/test/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationControllerTest.java:96]: Could use this instead:
`Collections.singletonList(new LateFilingPenalty())`

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java [hunk 9]
```diff
@@ -113,17 +126,15 @@ public String getConfirmation(@PathVariable String companyNumber,
 
     private String setUpPaymentDateDisplay(PayableLateFilingPenalty payableLateFilingPenalty) {
```
Comment (acallaghan-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:127]: Could this method be pulled out into the 'PenaltyUtils'? For me there is too much logic in the controller. I am hoping that we create a service layer for all this post the first release.

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java [hunk 10]
```diff
@@ -113,17 +126,15 @@ public String getConfirmation(@PathVariable String companyNumber,
...
+                    .format(DateTimeFormatter.ofPattern("d MMMM uuuu", Locale.UK));
         }
         return "";
     }
 
-    private String setUpPaymentAmountDisplay(PayableLateFilingPenalty payableLateFilingPenalty) {
-        if (payableLateFilingPenalty.getPayment() != null) {
-            return StringUtils.isEmpty(payableLateFilingPenalty.getPayment().getAmount()) ? "" :
-                    "£" + payableLateFilingPenalty.getPayment().getAmount() + " (no VAT is charged)";
+    private String setUpPaymentAmountDisplay(LateFilingPenalty payableLateFilingPenalty) {
```
Comment (acallaghan-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:135]: Could this method be pulled out into the 'PenaltyUtils'? For me there is too much logic in the controller. I am hoping that we create a service layer for all this post the first release.

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java (hunk 10 above)
Comment (klyttle-ch) [reply src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java]: Pulled both them methods out and put in penaltyUtils instead

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java [hunk 11]
```diff
@@ -49,13 +50,23 @@ public class ConfirmationController extends BaseController {
 
     private final SessionService sessionService;
 
+    @Autowired
```
Comment (rbull-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:53]: These 2 class variables should be `private final` like the others used in the Autowired Constructor below.
Please can you remove the `@Autowired` annotation on these 2 fields as it's handled by the constructor-arg Spring injection instead.

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java [hunk 12]
```diff
@@ -68,8 +79,9 @@ public String getConfirmation(@PathVariable String companyNumber,
                                   Model model) {
 
         Map<String, Object> sessionData = sessionService.getSessionDataFromContext();
+        List<LateFilingPenalty> lateFilingPenalties;
```
Comment (rbull-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:80]: Looks like these 2 variables should be moved down below inside the try block?
They can just be declared as `var` on lines 102 and 103.

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java [hunk 13]
```diff
@@ -68,8 +79,9 @@ public String getConfirmation(@PathVariable String companyNumber,
...
+        LateFilingPenalty lateFilingPenalty;
 
-        // Check that the session state is present
```
Comment (rbull-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:72]: Not sure why we need to remove these 3 existing comments in the code. Not really our concern yet?

File: src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java [hunk 14]
```diff
@@ -27,5 +32,20 @@ public String getFormattedOutstanding(final Integer outstandingAmount) {
...
+
+    public String setUpPaymentDateDisplay(PayableLateFilingPenalty payableLateFilingPenalty) {
+        if (payableLateFilingPenalty.getPayment() != null) {
+            return LocalDate.now()
+                    .format(DateTimeFormatter.ofPattern("d MMMM uuuu", Locale.UK));
+        }
+        return "";
+    }
+
+    public String setUpPaymentAmountDisplay(LateFilingPenalty payableLateFilingPenalty) {
```
Comment (rbull-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java:44]: Looks like this method variable should be renamed to be `lateFilingPenalty` to match the type as it's different from the other method?

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java (hunk 13 above)
Comment (klyttle-ch) [reply src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:72]: Added these back in

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java [hunk 15]
```diff
@@ -101,30 +113,13 @@ public String getConfirmation(@PathVariable String companyNumber,
             model.addAttribute(PENALTY_REF_ATTR, penaltyRef);
             model.addAttribute(COMPANY_NAME_ATTR, companyProfileApi.getCompanyName());
             model.addAttribute(REASON_ATTR, PENALTY_REASON);
-            model.addAttribute(PAYMENT_DATE_ATTR, setUpPaymentDateDisplay(payablePenalty));
-            model.addAttribute(PENALTY_AMOUNT_ATTR, setUpPaymentAmountDisplay(payablePenalty));
+            model.addAttribute(PAYMENT_DATE_ATTR, penaltyUtils.setUpPaymentDateDisplay(payablePenalty));
+            model.addAttribute(PENALTY_AMOUNT_ATTR, penaltyUtils.setUpPaymentAmountDisplay(lateFilingPenalty));
```
Comment (rbull-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:116]: It looks to me like we should rename this method from `PaymentAmount` to `PenaltyAmount` to keep in line with the model attribute terminology?

Since we appear to be using the following penalty-payment-api endpoint `GET /company/{company_number}/penalties/payable/{id}`

I'm finding it hard to follow what line 103 `getFirst` is. I think we should be getting the penalty amount via `transactions`. e.g. 

```
    "transactions": [
        {
            "transaction_id": "A0531369",
            "amount": 3000,
            "type": "penalty",
            "made_up_date": "2013-10-31"
        }
    ],
```

See [PayablePenaltyServiceImpl#getPayableLateFilingPenalty](https://github.com/search?q=repo%3Acompanieshouse%2Fpenalty-payment-web%20getPayableLateFilingPenalty&type=code)

* uk.gov.companieshouse.api.model.latefilingpenalty.PayableLateFilingPenalty#transactions
* uk.gov.companieshouse.api.model.latefilingpenalty.Transaction#amount WHERE transactionId IS `penaltyRef`

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java [hunk 16]
```diff
@@ -88,6 +97,8 @@ public String getConfirmation(@PathVariable String companyNumber,
         try {
             PayableLateFilingPenalty payablePenalty = payablePenaltyService
                     .getPayableLateFilingPenalty(companyNumber, payableRef);
+            List<LateFilingPenalty> lateFilingPenalties = penaltyPaymentService.getLateFilingPenalties(companyNumber, penaltyRef);
+            LateFilingPenalty lateFilingPenalty = lateFilingPenalties.getFirst();
```
Comment (rbull-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:101]: [Declare this local variable with "var" instead.](https://code-analysis.platform.aws.chdev.org/project/issues?id=uk.gov.companieshouse%3Apenalty-payment-web&issues=AZQm55fP1r4uIxLEm9YG&open=AZQm55fP1r4uIxLEm9YG&pullRequest=187)

File: src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java [hunk 17]
```diff
@@ -27,5 +32,20 @@ public String getFormattedOutstanding(final Integer outstandingAmount) {
...
+        if (payableLateFilingPenalty.getPayment() != null) {
+            return LocalDate.now()
+                    .format(DateTimeFormatter.ofPattern("d MMMM uuuu", Locale.UK));
+        }
+        return "";
+    }
+
+    public String setUpPaymentAmountDisplay(LateFilingPenalty lateFilingPenalty) {
+        if (lateFilingPenalty.getOriginalAmount() != null) {
+            return getFormattedOutstanding(lateFilingPenalty.getOriginalAmount());
```
Comment (rbull-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java:46]: Since this method looks like it is being used in 2 places now - `viewPenalties` for "outstanding" and `confirmationPage` for "originalAmount".  It would be better to rename it to be `getFormattedAmount`?

(Also since the `lateFilingPenalty.getOriginalAmount()` is done in 2 places inside this method, it should be extracted into a local variable assignment and used for better readability.)

My preference would be to remove this `setUpPaymentAmountDisplay` method and instead put the null handling in the other `getFormattedAmount` method?

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java (hunk 17 above)
Comment (klyttle-ch) [reply src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java]: I have refactored to getFormattedAmount in latest commit

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java [hunk 18]
```diff
@@ -101,30 +103,13 @@ public String getConfirmation(@PathVariable String companyNumber,
             model.addAttribute(PENALTY_REF_ATTR, penaltyRef);
             model.addAttribute(COMPANY_NAME_ATTR, companyProfileApi.getCompanyName());
             model.addAttribute(REASON_ATTR, PENALTY_REASON);
```
Comment (rbull-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:105]: It looks like we have 2 similar ways of showing the same "Late filing of accounts" reason for confirmationPage and viewPenalties pages.

Wondering, should we be trying to use the same utility method?  But thinking about it further here, would you be able to use the message property rather than the application.properties / static constants we have currently in the 2 places instead? So it would be similar to this comment I made for @klai-ch (https://github.com/companieshouse/penalty-payment-web/pull/188#discussion_r1900957696) where we would end up removing the util method `getViewPenaltiesLateFilingReason`.

* https://github.com/search?q=repo%3Acompanieshouse%2Fpenalty-payment-web+%22Late+filing+of+accounts%22&type=code
* https://github.com/search?q=repo%3Acompanieshouse%2Fpenalty-payment-web+%22reason%22&type=code

File: src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java [hunk 19]
```diff
@@ -27,5 +31,20 @@ public String getFormattedOutstanding(final Integer outstandingAmount) {
     public String getReferenceTitle(final String penaltyNumber) {
         return penaltyNumber.startsWith("A") ? "Reference Number" : "Penalty Reference";
     }
+
+    public String setUpPaymentDateDisplay(PayableLateFilingPenalty payableLateFilingPenalty) {
+        if (payableLateFilingPenalty.getPayment() != null) {
```
Comment (rbull-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java:36]: I don't think we even need this condition as when I looked in my "payable_resources" collection, I currently see the following:

```
        "payment" : {
            "amount" : "",
            "paid_at" : null,
            "reference" : "",
            "status" : "pending"
        },
```

So if we're not using the `"paid_at"` and "amount" from this payment object in the database, then just simplify it to show TODAY.

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java [hunk 20]
```diff
@@ -20,12 +24,24 @@ public String getViewPenaltiesLateFilingReason() {
...
     }
+
+    public String setUpPaymentDateDisplay() {
+        return LocalDate.now()
+                .format(DateTimeFormatter.ofPattern("d MMMM uuuu", Locale.UK));
+    }
+
+    public String setUpPaymentAmountDisplay(PayableLateFilingPenalty payableLateFilingPenalty) {
+        if (payableLateFilingPenalty.getPayment() != null) {
+            return getFormattedAmount(payableLateFilingPenalty.getTransactions().getFirst().getAmount());
```
Comment (rbull-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java:42]: It looks to me like we will need to pass in the "penaltyRef" from the path param and to be sure we are getting the correct amount, we should be doing a `filter` on the transactions if we're not using the "paid_at" and "amount" from this payment object in the database.

Given our confusion here with this payment date / amount stuff, for now should we just remove the `getPayment` condition here also? To simplify in the same way as https://github.com/companieshouse/penalty-payment-web/pull/187/files/d98aa7fcbad7aadb72ef2f10be2e98758acaea6c#r1901075510

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java (hunk 18 above)
Comment (klyttle-ch) [reply src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java]: Got this changed to come from messages.properties

File: src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java [hunk 21]
```diff
@@ -20,12 +24,21 @@ public String getViewPenaltiesLateFilingReason() {
         return viewPenaltiesLateFilingReason;
     }
 
-    public String getFormattedOutstanding(final Integer outstandingAmount) {
-        return OUTSTANDING_AMOUNT_FORMATTER.format(outstandingAmount);
+    public String getFormattedAmount(final Integer amount) {
+        return AMOUNT_FORMATTER.format(amount);
```
Comment (acallaghan-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java:30]: Just a preference thing but if this variable is only used in here, I would have it as a local variable.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java [hunk 22]
```diff
@@ -37,5 +41,14 @@ public String getLoginEmail(SessionService sessionService) {
         }
         return "";
     }
+
+    public String setUpPaymentDateDisplay() {
+        return LocalDate.now()
```
Comment (acallaghan-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java:46]: Can you add unit tests for these. Sonar is flagging it.

**Review by rbull-ch — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java (hunk 22 above)
Comment (klyttle-ch) [reply src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java:46]: Test cases have now been added and passing.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by dgordon-ch — APPROVED** [review]

**Review by acallaghan-ch — APPROVED** [review]
